models.py

```diff
@@ -1,33 +1,77 @@
-"""The model definitions."""
```

Comment:
Thanks a lot for making the code compatible with the new keras version and adding tests to the code :+1: